Documentation/ceph-debugging.md

```diff
@@ -0,0 +1,231 @@
+> applications that run on it. This document will use Kubernetes terms, and users are expected to be
+> empowered to look up Kubernetes information they do not already have. This document will connect
+> dots to explain how to use Kubernetes tools to get information needed in the Rook-Ceph context and
+> (when relevant) will briefly explain how Rook uses Kubernetes features.
+
+### Set operator log level to debug
+In general, the first place to look when encountering a failure is to get logs for the rook-ceph-operator pod. To get the best logs possible, set the operator log level to "DEBUG". There are many ways to do this, including modifying Helm's values.yaml or modifying the operator.yaml manifest. Regardless of method chosen, the log level can always be set by editing the deployment directly with kubectl.
+
+```sh
+kubectl --namespace rook-ceph edit deployment rook-ceph-operator
+# manually edit ROOK_LOG_LEVEL 'env' value "DEBUG" (more logs) or "INFO" (less logs) as desired
+```
+
+After editing the deployment, the operator pod will restart automatically and will start outputting
+logs with the new log level.
+
+If you are experiencing a particular failure, it may take some time for the Rook operator to reach
+the failure location again to report debug logs.
+
+### Use the toolbox pod
+Use the Rook toolbox pod to interface directly with the Ceph cluster via CLI. Enter into the toolbox
+pod as shown below.
+```
+kubectl --namespace rook-ceph exec -it deploy/rook-ceph-tools -- bash
+```
+
+If the rook-ceph-tools deployment does not exist, it should be created using the toolbox.yaml manifest.
+
+### Set Ceph log levels
+To set log levels for Ceph daemons, it is advised to use the Ceph CLI from the toolbox pod.
+
+### Get container logs
+Getting logs for a container in Kubernetes can be fairly easy, but to get logs with more clarity, it
+is slightly more complicated.
+
+#### Quick and dirty <!-- omit in toc -->
+At its simplest, get all the logs for all containers in a Pod as shown below. This is easy, but
+since Rook-Ceph pods almost always contain init containers in addition to running containers, the
+output does not make it clear which logs belong to makes debugging more difficult by not identifying
+which logs belong to which container in a Pod.
+```sh
+# Get logs for all containers in a Pod
+kubectl --namespace rook-ceph logs <pod-name> --all-containers
+```
+
+#### Manual, slow and more precise <!-- omit in toc -->
+To get logs for a single container, first determine which containers belong to a pod, then request
+logs only for the container desired.
+```sh
+# Get logs for a single container in a pod
+kubectl --namespace rook-ceph get pod <pod-name> -o jsonpath='{..initContainers[*].name} {..containers[*].name} {"\n"}'
+# ^ lists init containers followed by running containers
+kubectl --namespace rook-ceph logs <pod-name> --container <container-name>
+# ^ gets logs for a single container
+```
+
+#### Misc <!-- omit in toc -->
```

Comment:
These are standard K8s behavior. Could we just link to K8s docs instead of duplicating here?